src/pages/workspaces/workspace/Data.js

```diff
@@ -563,17 +565,41 @@ const WorkspaceData = _.flow(
     }
   }
 
-  const loadWdsSchema = async () => {
-    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
-      try {
-        setWdsSchema([])
-        setWdsSchemaError(undefined)
-        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(workspaceId)
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
-      }
+  const loadWdsSchemaAndUrl = async () => {
+    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd && isAzureWorkspace) {
+      await Ajax(signal).Apps.getV2AppInfo(workspaceId).then(async apps => {
+        try {
+          setWdsSchema([])
+          setWdsSchemaError(undefined)
+          const url = getWdsUrl(apps)
+          setWdsProxyUrl(url)
+          const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
+          setWdsSchema(wdsSchema)
+        } catch (error) {
+          setWdsSchemaError(error)
+        }
+      })
```

Comment:
Mixing `await` and `then` is a bit unconventional. This could be:

```suggestion
      const apps = await Ajax(signal).Apps.getV2AppInfo(workspaceId)
      try {
        setWdsSchema([])
        setWdsSchemaError(undefined)
        const url = getWdsUrl(apps)
        setWdsProxyUrl(url)
        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
        setWdsSchema(wdsSchema)
      } catch (error) {
        setWdsSchemaError(error)
      }
```